src/Build/BuildCheck/Infrastructure/ConfigurationProvider.cs

```diff
@@ -140,6 +136,33 @@ public static void CheckCustomConfigurationDataValidity(string projectFullPath,
         return configurations;
     }
 
+    internal Dictionary<string, string> GetConfiguration(string projectFullPath, string ruleId)
+    {
+        var config = new Dictionary<string, string>();
+        try
+        {
+            config = s_editorConfigParser.Parse(projectFullPath);
```

Comment:
This appears to be called in a loop - for each analyzer, for each rule -  and it is an expensive operation. We are basically initializing a dictionary with _all_ rules applicable to the given project file over and over again. Would it make sense to make this call only once for the given project file, filter the result to rules starting with `build_check.`, and then for each analyzer / for each rule search this smaller cached dictionary?